packages/editor/src/plugins/solution/renderer.tsx

```diff
@@ -27,57 +25,72 @@ export function SolutionRenderer({
 }: SolutionRendererProps) {
   const { strings } = useInstanceData()
 
-  const [solutionVisible, setSolutionVisible] = useState(solutionVisibleOnInit)
+  const [visible, setVisible] = useState<false | 'solution' | 'strategy'>(
+    solutionVisibleOnInit ? 'solution' : false
+  )
 
   return (
     <>
-      <div className="flex">
-        {renderSolutionToggle()}
+      <nav>
+        {hideToggle ? null : (
+          <>
+            {strategy ? renderToggle('strategy') : null}{' '}
+            {renderToggle('solution')}
+          </>
+        )}
         {elementAfterToggle}
-      </div>
-      {solutionVisible && renderSolutionContent()}
+      </nav>
+      {visible ? (
+        <div className="serlo-solution-box">
+          {visible === 'solution' ? renderSolutionContent() : null}
+          {visible === 'strategy' && strategy ? strategy : null}
+        </div>
+      ) : null}
```

Comment:
If `visible === 'strategy' && strategy === null`, this `div` will be rendered empty. Is that the desired outcome?

Reply:
You are correct, but if `strategy === null` the toggle is not shown so this can't happen.
Do you have a simple idea to make this cleaner?
I can't think of something right now where I think it's worth the extra complexity.

I will simplify `{visible === 'strategy' && strategy ? strategy : null}` to `{visible === 'strategy' ? strategy : null}` because it does the same.

Reply:
Nah, sounds good, I didn't get that context when commenting, thanks for explaining.